src/main/java/edu/cornell/kfs/fp/service/impl/RecurringDisbursementVoucherDocumentServiceImpl.java

```
@@ -409,8 +417,117 @@ public Object call() throws WorkflowException {
+                    dv = null;
+                }
+
+                if (dv != null) {
+                    if (isKfsSystemUserDvInitiator(dv)) {
+                        try {
+                            addNoteToAutoApproveDv(dv, "Batch job Submit and Blanket Approve performed for DV spawned by Recurring DV.");
+                            try{
+                                blanketApproveDisbursementVoucherDocument(dv);
+                                approvalCount++;
+
+                            } catch (WorkflowException e) {
+                                errorCount++;
+                            }
+                        } catch (WorkflowException e) {
+                            errorCount++;
+                        }
+                    }
+                    else {
+                        LOG.error("Detected document initiator was not KFS System user. Auto blanket approval was NOT attemtped for document ID: " + dv.getDocumentNumber());
+                        errorCount++;
+                    }
+                }
+            }
+        }
+        else {
+            LOG.info("No DV's spwned by Recurring DV found. Nothing will be auto approved. ");
+        }
+        LOG.info("Number of Disbursement Vouchers that generated Errors:: " + errorCount);
+        LOG.info("Number of Disbursement Vouchers successfuly Blanket Approved (fully processed):: " + approvalCount);
+        LOG.info("autoApproveDisbursementVouchersSpawnedByRecurringDvs: Leaving.");
+        return errorCount == 0;
+    }
     
-    
+    private boolean isKfsSystemUserDvInitiator(CuDisbursementVoucherDocument dv) {
+        String dvDocInitatorPrincipalId = dv.getDocumentHeader().getWorkflowDocument().getInitiatorPrincipalId();
+
+        Person kfsSystemUser = getPersonService().getPersonByPrincipalName(KFSConstants.SYSTEM_USER);
+        String kfsSystemUserPrincipalId = kfsSystemUser.getPrincipalId();
```

If kfsSystemUserPrincipalId is null you probably want to throw an error, that means something went wrong, user does not exist? Not approving a DV because this is null does not seem like the right course of action